crates/runtime-configs/src/lib.rs

```diff
@@ -194,19 +200,43 @@ pub async fn set(
         ConfigsImplementor::EnvVars => Ok(EnvVars::set(key, value)?),
         ConfigsImplementor::UserSecrets => Ok(UserSecrets::set(key, value, toml_file_path)?),
         ConfigsImplementor::AzApp => Ok(AzApp::set(key, value).await?),
+        _ => panic!("unknown configuration type"),
     }
 }
 
 pub async fn get_from_state(config_name: &str, state: &BasicState) -> Result<String> {
+    let c = state
+        .configs_map
+        .as_ref()
+        .expect("this capability needs a [capability.configs] section...")
+        .get(config_name)
+        .unwrap_or_else(|| panic!("failed to get config '{}'", config_name));
+
+    let (store, name) = maybe_get_config_store_and_value(c)?;
+
     let config = String::from_utf8(
-        get(&state.secret_store, config_name, &state.slightfile_path)
+        get(&store, &name, &state.slightfile_path)
             .await
             .with_context(|| {
                 format!(
-                    "failed to get '{}' secret using secret store type: {}",
-                    config_name, state.secret_store
+                    "failed to get '{}' secret using secret store type: '{}'",
+                    config_name, store
                 )
             })?,
     )?;
     Ok(config)
 }
+
+fn maybe_get_config_store_and_value(c: &str) -> Result<(String, String)> {
```

Comment:
This function is pretty hard to understand. Could you write some unittests for it?

Reply:
done ✅